test/parallel/test-https-agent-jssocket-close.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const tls = require('tls');
+const fixtures = require('../common/fixtures');
+const https = require('https');
+const key = fixtures.readKey('agent1-key.pem');
+const cert = fixtures.readKey('agent1-cert.pem');
+const net = require('net');
+const { Duplex } = require('stream');
+
```

Comment:
What this test is testing for is fairly nuanced. A comment describing the expected behavior would help those coming along later understand what's happening here.